### ISSUE TITLE
range queries don't respect maxStalenessInterval 

### ISSUE THREAD

**Issue by propertone** [VictoriaMetrics/VictoriaMetrics#784]
**Describe the bug**
A range query does not appear to respect the `-search.maxStalenessInterval` option. 

**To Reproduce**
Sample data:

```
{"metric": {"__name__": "test", "var": "0"}, "values": [1], "timestamps": [1600476000000]}
{"metric": {"__name__": "test", "var": "1"}, "values": [1], "timestamps": [1600476030000]}
{"metric": {"__name__": "test", "var": "2"}, "values": [1], "timestamps": [1600476060000]}
{"metric": {"__name__": "test", "var": "3"}, "values": [1], "timestamps": [1600476090000]}
{"metric": {"__name__": "test", "var": "4"}, "values": [1], "timestamps": [1600476120000]}
```
Run victoriametrics with `-search.maxStalenessInterval 30s`

**Expected behavior**
Range query of `sum(test)` should give the same result as evaluating an instant query at every step, however the result varies depending on the step, despite `-search.maxStalenessInterval` being set.

For step=30s, the values are [1,1,1,1,1]. For step=120s, the values are [1,4], but should be [1,1].

**Version**
```
1.40.0
```

**Used command-line flags**
```
victoria-metrics -httpListenAddr :8429 -retentionPeriod 1 -storageDataPath /var/tmp/vm-test -selfScrapeInterval 30s -search.maxStalenessInterval 30s
```

**Additional context**
From #748

**Comment by valyala** [comment]
The bug must be fixed in the commit c584aece388fa46cd2eecd6080fdd9e70d84325e . @propertone , could you build VictoriaMetrics from this commit according to [these docs](https://victoriametrics.github.io/#how-to-build-from-sources) and verify whether `-search.maxStalenessInterval` command-line flag works there as expected?

**Comment by propertone** [comment]
I tested this out and can confirm that it fixes the problem. Thanks for the quick fix!

**Comment by valyala** [comment]
The bugfix has been included in [v1.42.0 release](https://github.com/VictoriaMetrics/VictoriaMetrics/releases). Closing the issue as fixed.
